equational_theories/MemoFinOp.lean

```diff
@@ -9,39 +9,41 @@ namespace MemeFinOp
 
 open Lean Meta Elab Term
 
-def opOfTable {n : Nat} (t : Nat) (a : Fin n) (b : Fin n) : Fin n :=
-  let i := a.val * n + b.val
-  let r := (t / n^i) % n
-  ⟨r, Nat.mod_lt _ (Fin.pos a)⟩
+def opOfTable {n : Nat} (s t : Nat) (a : Fin n) (b : Fin n) : Fin n :=
+  ⟨(t >>> (s * (a.val * n + b.val))) % n, Nat.mod_lt _ (Fin.pos a)⟩
 
 /-- `enum n` is the array of all elements of `Fin n` in order -/
-private def Fin.enum (n) : Array (Fin n) := Array.ofFn id
+private def Fin.enum (n) : Array (Fin n) := Array.ofFn Fin.rev
```

Comment:
You probably should update the docstring as well (or remove it for this private helper)

Reply:
I undid this change, indeed it didn't help